yw_website/yw_website/apps/website/serializers.py

```diff
@@ -61,3 +66,102 @@ class TagFileSerializer(serializers.ModelSerializer):
     class Meta:
         model = TagFile
         fields = '__all__'
+
+class YesWorkflowSaveSerializer(serializers.ModelSerializer):
+    model = serializers.CharField(required=False, allow_blank=False)
+    model_checksum = serializers.CharField(required=False, allow_blank=True, max_length=128)
+    graph = serializers.CharField(required=False, allow_blank=False)
+    recon = serializers.CharField(required=False, allow_blank=True)
+    tags = serializers.ListField(
+        child=serializers.CharField(required=False, allow_blank=True, max_length=32)
+    )
+    scripts = ScriptSerializer(many=True)
+    files = FileSerializer(many=True)
+
+    class Meta:
+        model = Workflow
+        fields =('title','description','model','model_checksum', 'graph', 'recon', 'tags', 'scripts', 'files' )
+
+
+    def create(self, validated_data):
+        w = Workflow(
+            user=validated_data.get('username'),
+            title=validated_data.get('title'),
+            description=validated_data.get('description')
+        )
+        w.save()
+        v = Version(
+            workflow=w,
+            yw_model_checksum=validated_data.get('model_checksum'),
+            yw_model_output=validated_data.get('model'),
+            yw_graph_output=validated_data.get('graph'),
+            last_modified=datetime.datetime.now(tz=timezone.utc)
+        )
+        v.save()
+        r = Run(
+            version=v,
+            run_time_stamp=datetime.datetime.now(tz=timezone.utc),
+            yw_recon_output=validated_data.get('recon')
+        )
+        r.save()
+
+        tags = TagSerializer(validated_data.get('tags'), many=True)
+        for tag in tags.data:
+            t = Tag(parent_tag=None, tag_type=Workflow, title=tag.get('title'))
+            t.save()
+
+        scripts = ScriptSerializer(validated_data.get('scripts'), many=True)
+        for script in scripts.data:
+            s = Script(name=script.get('name'), version=v, checksum=script.get('checksum'), content=script.get('content'))
+            s.save()
+
+        files = FileSerializer(validated_data.get('files'), many=True)
+        for file in files.data:
+            f = File(checksum=file.get('checksum'), size=file.get('size'), name=file.get('name'), uri=file.get('uri'), last_modified=file.get('last_modified'))
+            f.save()
+
+        return w.pk, v.pk, r.pk
+    
+    def update(self, validated_data):
+        w = Workflow.objects.get(pk=self.context.get('workflow_id'))
```

Comment:
are we checking to make sure a workflow is associated with a particular user and returning and error if not?

Reply:
No will add